specification/billing/resource-manager/Microsoft.Billing/preview/2020-12-15-privatepreview/payNow.json

```diff
@@ -0,0 +1,502 @@
+        }
+      }
+    },
+    "InitiatePayNowProperties": {
+      "type": "object",
+      "description": "Request parameters to initiate pay now.",
+      "properties": {
+        "invoiceName": {
+          "type": "string",
+          "description": "The ID that uniquely identifies an invoice which is being paid off."
+        },
+        "payOffBillingAccountName": {
+          "type": "string",
+          "description": "The ID of the billing account whose pre-payment balance will be used to pay off the overage.. Currently only within billing account pay-off is supported."
+        },
+        "payOffDate": {
+          "type": "string",
+          "description": "The date used to get the pre-payment balance from pay off billing account. The date should be specified in MM-DD-YYYY format."
+        }
+      }
+    },
+    "PayNowDetails": {
+      "type": "object",
+      "description": "Details of the pay now.",
+      "properties": {
+        "properties": {
+          "description": "Details of the pay now.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/PayNowProperties"
+        }
+      }
+    },
+    "PayNowProperties": {
+      "type": "object",
+      "description": "Request parameters to initiate pay now.",
+      "properties": {
+        "createdTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The time at which the pay now request was created.",
+          "readOnly": true
+        },
+        "errorCode": {
+          "description": "The error code for failed pay now.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "errorMessage": {
+          "description": "The error message for failed pay now.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "id": {
+          "description": "The pay now operation id for the initiated pay now.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "initiatorEmailAddress": {
+          "description": "The email ID of the user who initiated the pay now request.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "invoiceName": {
+          "description": "The invoice number being paid off.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "overageAmount": {
+          "description": "The total pre-payment eligible overage amount.",
+          "readOnly": true,
+          "$ref": "#/definitions/Amount"
+        },
+        "payOffBillingAccountName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The ID of the billing account whose pre-payment balance will be used to pay off the overage."
+        },
+        "payOffDate": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The effective date used to get the pre-payment balance from pay off billing account."
+        },
+        "percentComplete": {
```

Comment:
How is this used? And why is it a string?

Reply:
This is used to denote the process completion status. PayNow is a bigger work flow and requires many systems to update including invoices. So in order to know, how much percentage of the task is completed, we have a field to mark as PercentComplete. I will get back on the data type as string.